searcher.rb

```diff
@@ -0,0 +1,62 @@
+require 'rubygems'
+require 'bundler/setup'
+require 'mechanize'
+require_relative 'result'
+
+class Searcher
+  attr_reader :results
+  def initialize(opts={})
+    @results = []
+  end
+
+  def search(opts={})
+    run_search(opts)
+    parse_results
+  end
+
+  def run_search(opts)
+    for_one = opts[:for_one] ? opts[:for_one] : ''
```

Comment:
Another way to write this:
`opts[:for_one] ||= ''`
and then use `opts[:for_one]` in `m.get()`. You can repeat the same for other options too.